cupyx/scipy/interpolate/_interpolate.py

```diff
@@ -0,0 +1,977 @@
+    # Determine if the breakpoints are in ascending order or descending one
+    ascending = (x[x.shape[0] - 1] >= x[0]).item()
+
+    intervals = cupy.empty(xp.shape, dtype=cupy.int64)
+    interval_kernel = INTERVAL_MODULE.get_function('find_breakpoint_position')
+    interval_kernel(((xp.shape[0] + 128 - 1) // 128,), (128,),
+                    (x, xp, intervals, extrapolate, xp.shape[0], x.shape[0],
+                     ascending))
+
+    # Compute coefficient displacement stride (in elements)
+    c_shape = cupy.asarray(c.shape, dtype=cupy.int64)
+    c_strides = cupy.asarray(c.strides, dtype=cupy.int64) // c.itemsize
+
+    ppoly_kernel = _get_module_func(PPOLY_MODULE, 'eval_ppoly', c)
+    ppoly_kernel(((xp.shape[0] + 128 - 1) // 128,), (128,),
+                 (c, x, xp, intervals, dx, c_shape, c_strides,
+                  xp.shape[0], out))
+
+
+def _fix_continuity(c, x, order):
+    """
+    Make a piecewise polynomial continuously differentiable to given order.
+
+    Parameters
+    ----------
+    c : ndarray, shape (k, m, n)
+        Coefficients local polynomials of order `k-1` in `m` intervals.
+        There are `n` polynomials in each interval.
+        Coefficient of highest order-term comes first.
+
+        Coefficients c[-order-1:] are modified in-place.
+    x : ndarray, shape (m+1,)
+        Breakpoints of polynomials
+    order : int
+        Order up to which enforce piecewise differentiability.
+    """
+    # Compute coefficient displacement stride (in elements)
+    c_shape = cupy.asarray(c.shape, dtype=cupy.int64)
+    c_strides = cupy.asarray(c.strides, dtype=cupy.int64) // c.itemsize
+
+    continuity_kernel = _get_module_func(PPOLY_MODULE, 'fix_continuity', c)
+    continuity_kernel((1,), (1,),
+                      (c, x, order, c_shape, c_strides, x.shape[0]))
+
+
+def _integrate(c, x, a, b, extrapolate, out):
+    """
+    Compute integral over a piecewise polynomial.
+
+    Parameters
+    ----------
+    c : ndarray, shape (k, m, n)
+        Coefficients local polynomials of order `k-1` in `m` intervals.
+    x : ndarray, shape (m+1,)
+        Breakpoints of polynomials
+    a : double
+        Start point of integration.
+    b : double
+        End point of integration.
+    extrapolate : bool, optional
+        Whether to extrapolate to out-of-bounds points based on first
+        and last intervals, or to return NaNs.
+    out : ndarray, shape (n,)
+        Integral of the piecewise polynomial, assuming the polynomial
+        is zero outside the range (x[0], x[-1]).
+        This argument is modified in-place.
+    """
+    # Determine if the breakpoints are in ascending order or descending one
+    ascending = (x[x.shape[0] - 1] >= x[0]).item()
+
+    a = cupy.asarray([a], dtype=cupy.float64)
+    b = cupy.asarray([b], dtype=cupy.float64)
+    if not ascending:
+        a, b = b, a
+
+    start_interval = cupy.empty(a.shape, dtype=cupy.int_)
+    end_interval = cupy.empty(b.shape, dtype=cupy.int_)
```

Comment:
The kernel takes `long long*` argument, but `cupy.int_` is 32bit on windows.
```suggestion
    start_interval = cupy.empty(a.shape, dtype=cupy.int64)
    end_interval = cupy.empty(b.shape, dtype=cupy.int64)
```